src/controllers/tests/validate.controller.test.ts

```diff
@@ -0,0 +1,213 @@
+import request from 'supertest';
+
+import { App } from '../../app';
+import { ProblemException } from '../../exceptions/problem.exception';
+
+import { ValidateController } from '../validate.controller';
+
+const validJSONAsyncAPI = {
+  asyncapi: '2.0.0',
+  info: {
+    title: 'My API',
+    version: '1.0.0',
+  },
+  channels: {
+    '/test/tester': {
+      subscribe: {
+        message: {
+          payload: {
+            type: 'object',
+            properties: {
+              name: {
+                type: 'string',
+                'x-parser-schema-id': '<anonymous-schema-2>',
+              },
+            },
+            'x-parser-schema-id': '<anonymous-schema-1>',
+          },
+          'x-parser-original-schema-format':
+            'application/vnd.aai.asyncapi;version=2.0.0',
+          'x-parser-original-payload': {
+            type: 'object',
+            properties: {
+              name: {
+                type: 'string',
+              },
+            },
+          },
+          schemaFormat: 'application/vnd.aai.asyncapi;version=2.0.0',
+          'x-parser-message-parsed': true,
+          'x-parser-message-name': '<anonymous-message-1>',
+        },
+      },
+    },
+  },
+  'x-parser-spec-parsed': true,
+};
+const validYAMLAsyncAPI = `
+asyncapi: '2.2.0'
+info:
+  title: Account Service
+  version: 1.0.0
+  description: This service is in charge of processing user signups
+channels:
+  user/signedup:
+    subscribe:
+      message:
+        $ref: '#/components/messages/UserSignedUp'
+components:
+  messages:
+    UserSignedUp:
+      payload:
+        type: object
+        properties:
+          displayName:
+            type: string
+            description: Name of the user
+          email:
+            type: string
+            format: email
+            description: Email of the user
+`;
+const invalidJSONAsyncAPI = {
+  asyncapi: '2.0.0',
+  info: {
+    tite: 'My API', // spelled wrong on purpose to throw an error in the test
+    version: '1.0.0'
+  },
+  channels: {}
+};
+
+describe('ValidateController', () => {
+  describe('[POST] /validate', () => {
+    it('should validate AsyncAPI document in JSON', async () => {
+      const app = new App([new ValidateController()]);
+
+      return await request(app.getServer())
+        .post('/validate')
+        .send({
+          asyncapi: validJSONAsyncAPI,
+          template: '@asyncapi/html-template',
+          parameters: {
+            version: '2.1.37',
+          }
```

Comment:
Please remove all occurrences of `template` and `parameters` fields, because they are only related to the `/generator` path :) `/validate` should have only `asyncapi` field (I also wondering if we should go without that field and treat whole `req.body` as `asyncapi` field) 🤔

Reply:
thanks for pointing this out, I forgot to put this question on the PR.
I think I had troubles because `requestBodyValidationMiddleware` said these were required, but I couldn't exactly point out where this template field was being set to required.

I agree with you and will remove the `template` and `parameters` fields 👍

Reply:
Yes, this middleware reads the required fields from the OpenAPI specification and does the validation based on that :)